tests/lax_numpy_test.py

```diff
@@ -2735,18 +2735,21 @@ def testPrecision(self):
 
   @parameterized.named_parameters(
       jtu.cases_from_list(
-        {"testcase_name": ("_shape={}_axis={}_dtype={}").format(shape, axis, dtype),
+        {"testcase_name": ("_shape={}_varargs={} axis={}_dtype={}")
+         .format(shape, varargs, axis, dtype),
          "shape": shape,
+         "varargs": varargs,
          "axis": axis,
          "dtype": dtype, "rng_factory": rng_factory}
         for shape in [(10,), (10, 15), (10, 15, 20)]
-        for _num_axes in range(len(shape)) 
+        for _num_axes in range(len(shape))
+        for varargs in [0, (1,), (2, 0)]
         for axis in itertools.combinations(range(len(shape)), _num_axes)
         for dtype in inexact_dtypes
         for rng_factory in [jtu.rand_default]))
-  def testGradient(self, shape, axis, dtype, rng_factory):
+  def testGradient(self, shape, varargs, axis, dtype, rng_factory):
     rng = rng_factory()
-    args_maker = self._GetArgsMaker(rng, [shape], [dtype])
+    args_maker = self._GetArgsMaker(rng, [shape], [dtype], [varargs])
```

Comment:
I'm not sure this call to `_GetArgsMaker` is right: the signature is `_GetArgsMaker(self, rng, shapes, dtypes, onp_arrays=True)`, so the last argument is just meant to be a bool.

Maybe instead `varargs` should be passed in the lambdas below, like

```python
    jnp_fun = lambda y: jnp.gradient(y, varargs, axis=axis)
    onp_fun = lambda y: onp.gradient(y, varargs, axis=axis)
```

WDYT?

Reply:
Ah, thanks for the clarification! 
It definitely makes sense - sorry I struggled to get the logic of the `_GetArgsMaker`, much clearer now.
I just committed that - tests are ok locally, waiting for the ci.

Reply:
CI failing, it seems like the random permutation between different inputs cause `len(varargs)` to be different from `len(axis)`. What do you think? Do you forsee any other problem?

Reply:
I haven't looked into your particular problem here, but if you want `varargs` and `axis` to have equal lengths, you must change the test harness to ensure that (e.g., first generate one, and then generate the other to have the same length.)

Reply:
Thanks @hawkinsp - just added that in a2fcad9. `len(varargs)` now returns `num_axes`